src/features/ProfilePage/components/PasswordEditor.tsx

```diff
@@ -35,12 +43,27 @@ const PasswordEditor = () => {
     newPassword !== repeatedPassword;
 
   const isSavingDisabled =
-    !currentPassword.length || isPasswordTooShort || arePasswordsNotMatching;
+    isLoading ||
+    !currentPassword.length ||
+    isPasswordTooShort ||
+    arePasswordsNotMatching;
```

Comment:
For me this file looks too "statey", but it might be just so that its unavoidable.
I would maybe try to abstract some logic to its own hook, maybe return a state like

```typescript
type PasswordInput = {
isDirty: boolean
isOk: boolean
}
type PasswordFormState = {
first: PasswordInput
second: PasswordInput
}
```

Secondly I would parse the password for allowed letters. Maybe this regex from here https://github.com/sos-lapsikyla/ylitse-app/blob/master/src/lib/validators.ts#L85 could be a good one:



```typescript
const isValidPassword = (value: string): boolean => {
  return /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[()#?!@$%^&*-]).{8,}$/.test(value);
};

// it could be even more parsed:
export const ValidPassword = new t.Type<string, string, unknown>(
  'ValidPassword',
  (input: unknown): input is string => typeof input === 'string',
  (input, context) =>
    typeof input === 'string' && isValidPassword(input)
      ? t.success(input)
      : t.failure(input, context),
  t.identity,
);

// simple usage
 const passwordParsingResult = ValidPassword.decode(newPassword);
 const isValidPassword = isRight(passwordParsingResult);
  ```


But maybe it looks a bit better without the effects already.
I would do refactoring first and then think about abstracting some logic away a bit.

Reply:
And also it would be great to implement unit-tests against password-validator

Reply:
@kuosmark so about this? in the next PR then i guess?

Reply:
I made a [new backlog task](https://trello.com/c/i9mE16Ry/986-ui-password-validation) for this. I have some questions about this we can go through in an upcoming meeting.